src/server/http/__tests__/version_check.js

```diff
@@ -0,0 +1,76 @@
+import expect from 'expect.js';
+import { fromNode } from 'bluebird';
+import { resolve } from 'path';
+import * as kbnTestServer from '../../../../test/utils/kbn_server';
+
+const src = resolve.bind(null, __dirname, '../../../../src');
+
+const versionHeader = 'kbn-version';
+const version = require(src('../package.json')).version;
+
+describe('version_check request filter', function () {
+  function makeRequest(kbnServer, opts) {
```

Comment:
I'd love to replace the `kbnTestServer.makeRequest()` function with something like this. Or update it to use `kbnServer.inject()` which already promisifies the `kbnServer.server.inject()`.